assets/js/modules/idea-hub/components/dashboard/DashboardIdeasWidget/index.js

```diff
@@ -0,0 +1,141 @@
+/**
+ * DashboardIdeasWidget component
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import Tab from '@material/react-tab';
+import TabBar from '@material/react-tab-bar';
+import { useHash, useMount } from 'react-use';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useState, useRef, useCallback } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import whenActive from '../../../../../util/when-active';
+import NewIdeas from './NewIdeas';
+import SavedIdeas from './SavedIdeas';
+import DraftIdeas from './DraftIdeas';
+
+const getHash = ( hash ) => hash ? hash.replace( '#', '' ) : false;
+const isValidHash = ( hash ) => getHash( hash ) in DashboardIdeasWidget.tabToIndex;
+
+const DashboardIdeasWidget = () => {
+	const ideaHubContainer = useRef();
+	const [ hash, setHash ] = useHash();
```

Comment:
I know that IB says that we need to use hashes, but let's use `core/ui` store to control which tab is open instead of hashes. We use the same approach for other widgets as well, for example, PageSpeed Insights uses it for its tabs: https://github.com/google/site-kit-wp/blob/cf3e8916594c9cb43288f4a639ec938ed53f3a63/assets/js/modules/pagespeed-insights/components/dashboard/DashboardPageSpeed.js#L62-L63

Reply:
@eugene-manuilov The thing is that we need to be able to link to a specific tab via the URL as per the ticket description:
> It should be possible to link to a specific tab in the widget.

Unless you want to say to not use the `activeTabIndex` state variable and use the `core/ui` store see what tab is opened. If this is the case, I think it's overkilled in this scenario unless am missing something? 🤔 
I checked the PageSpeed Insights and I can't find a way where you can specify a tab to be open.
Let me know :)

Reply:
Ok, let it be as is then.